crates/polars-core/src/datatypes/dtype.rs

```diff
@@ -940,6 +940,49 @@ impl DataType {
         }
         level
     }
+
+    pub fn byte_size(&self) -> Option<usize> {
+        match self {
+            DataType::Boolean => Some(1),
```

Comment:
Possibly boolean is bit-packed? Validate, and if so this presumably won't work and needs to be `None`.

Reply:
Seems like it per https://arrow.apache.org/docs/cpp/api/datatype.html, and... it's not tested. So also need to probably add more tests, and also look at coverage results and take that into account.